### ISSUE TITLE
bug: checkly_check_group_v2 migration leads to monitors destruction

### ISSUE THREAD

**Issue by abuisine** [checkly/terraform-provider-checkly#360]
### Terraform version

OpenTofu v1.11.5

### Provider version

v1.21.1

### Steps to reproduce

While migrating my plan from `checkly_check_group` to `checkly_check_group_v2` I faced a bunch of `404` errors on url monitors.
I realised that Terraform/Tofu destroyed the `checkly_check_group` object which triggered a cascade deletion of all affiliated monitors.
Terraform/Tofu was therefore not able to modify monitors in order to change their `group_id`

### What is expected?

Deletion of `checkly_check_group` should not trigger cascade deletion of related monitors

### What is actually happening?

Monitors (url, icmp, dns, ...) are deleted rather than modified

### System Info

```shell

```

**Comment by guolau** [comment]
Hi @abuisine, I wanted to give you a quick update. 

I spoke with our team and learned that this is due to how our API is implemented. The API that the Terraform provider uses will delete all checks within a group if that group is deleted. So this is not a fault of the Terraform provider exactly, but is nonetheless an issue for us to resolve.

I'm sorry for the trouble here with the deleted monitors. I will follow up internally with the team regarding the API issue so we can prevent this going forward.
